evm/v0.5/test/PrepaidAggregator_test.js

```diff
@@ -0,0 +1,627 @@
+      it('reverts', async () => {
+        for (let i = 0; i < limit; i++) {
+          await aggregator.addOracle(web3.utils.randomHex(20), {
+            from: personas.Carol,
+          })
+        }
+        await h.assertActionThrows(async () => {
+          await aggregator.addOracle(personas.Neil, { from: personas.Carol })
+        })
+      })
+    })
+  })
+
+  describe('#removeOracle', async () => {
+    beforeEach(async () => {
+      await aggregator.addOracle(personas.Neil, { from: personas.Carol })
+    })
+
+    it('decreases the oracle count', async () => {
+      const pastCount = await aggregator.oracleCount.call()
+      await aggregator.removeOracle(personas.Neil, { from: personas.Carol })
+      const currentCount = await aggregator.oracleCount.call()
+
+      assertBigNum(currentCount, pastCount.sub(h.bigNum(1)))
+    })
+
+    it('updates the answer range', async () => {
+      const pastMin = await aggregator.minAnswerCount.call()
+      const pastMax = await aggregator.maxAnswerCount.call()
+
+      await aggregator.removeOracle(personas.Neil, { from: personas.Carol })
+
+      assertBigNum(
+        pastMin.sub(h.bigNum(1)),
+        await aggregator.minAnswerCount.call(),
+      )
+      assertBigNum(
+        pastMax.sub(h.bigNum(1)),
+        await aggregator.maxAnswerCount.call(),
+      )
+    })
+
+    context('when the oracle is not currently added', async () => {
+      beforeEach(async () => {
+        await aggregator.removeOracle(personas.Neil, { from: personas.Carol })
+      })
+
+      it('reverts', async () => {
+        await h.assertActionThrows(async () => {
+          await aggregator.removeOracle(personas.Neil, { from: personas.Carol })
+        })
+      })
+    })
+
+    context('when called by anyone but the owner', async () => {
+      it('reverts', async () => {
+        await h.assertActionThrows(async () => {
+          await aggregator.removeOracle(personas.Neil, { from: personas.Ned })
+        })
+      })
+    })
+
+    context('when called with a lower number of minimum oracles', async () => {
+      beforeEach(async () => {
+        await aggregator.setAnswerCountRange(0, 0, {
+          from: personas.Carol,
+        })
+      })
+
+      it('does not lower the answer range below 0', async () => {
+        await aggregator.removeOracle(personas.Neil, { from: personas.Carol })
+
+        assert.equal(0, await aggregator.minAnswerCount.call())
+        assert.equal(0, await aggregator.maxAnswerCount.call())
+      })
+    })
+  })
+
+  describe('#withdrawFunds', () => {
+    context('when called by the owner', () => {
+      it('succeeds', async () => {
+        await aggregator.withdrawFunds(personas.Carol, deposit, {
+          from: personas.Carol,
+        })
+
+        assertBigNum(0, await aggregator.availableFunds.call())
+        assertBigNum(deposit, await link.balanceOf.call(personas.Carol))
+      })
+
+      it('does not let withrawals happen multiple times', async () => {
```

Comment:
withDrawals

Reply:
Strictly speaking, this is failing because Carol's tried to overdraw, not because she's withdrawn multiple times, right?

Reply:
Correct. Maybe the test is unnecessary?

Reply:
I think it's valuable, it's just that the `it('description', ...)` is inaccurate.